chainer/functions/array/repeat.py

```diff
@@ -0,0 +1,128 @@
+import six
+
+from chainer import cuda
+from chainer import function_node
+from chainer.utils import type_check
+
+
+class Repeat(function_node.FunctionNode):
+
+    """Repeat elements of an array."""
+
+    def __init__(self, repeats, axis=None):
+        if isinstance(repeats, six.integer_types):
```

Comment:
This condition matches if `repeats` is `bool`. Please exclude such cases.
Also, please test such cases in the unit test.

Reply:
Actually NumPy/CuPy accepts `bool` for `repeats`. Should it be more strict than backend libraries?

```
>>> np.repeat(np.array([1.0, 2.0, 3.0]), True)
array([ 1.,  2.,  3.])
>>> np.repeat(np.array([1.0, 2.0, 3.0]), [False, True, 2])
array([ 2.,  3.,  3.])
```

Reply:
Oh... thank you for confirmation. So how about testing those inputs as well?

Reply:
I agree, I added tests for bools.